metrics/ci_integrations/test/destination/firestore/adapter/firestore_destination_client_adapter_test.dart

```diff
@@ -273,14 +362,267 @@ void main() {
+        await adapter.fetchBuildsWithStatus(
+          testProjectId,
+          buildStatus,
+        );
+
+        verify(_collectionReferenceMock.getDocuments()).called(once);
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() returns the builds with requested status",
+      () async {
+        when(_documentMock.map).thenReturn(buildDataTestJson);
+        whenFetchBuildsWithStatus().thenAnswer(
+          (_) => Future.value([_documentMock]),
+        );
+
         final result = await adapter.fetchBuildsWithStatus(
-          'id',
-          BuildStatus.unknown,
+          testProjectId,
+          BuildStatus.successful,
+        );
+
+        expect(result, equals([buildData]));
+      },
+    );
+
+    test(
+      ".fetchBuildsWithStatus() throws a DestinationError if reading documents fails",
+      () {
+        whenFetchBuildsWithStatus().thenAnswer(
+          (_) => Future.error(firestoreException),
         );
 
-        expect(result, isEmpty);
+        final result = adapter.fetchBuildsWithStatus(
+          testProjectId,
+          BuildStatus.successful,
+        );
+
+        expect(result, throwsDestinationError);
+      },
+    );
+
+    test(
+      ".updateBuilds() throws an ArgumentError if the given builds is null",
+      () {
+        expect(
+          () => adapter.updateBuilds(testProjectId, null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      ".updateBuilds() throws a DestinationError if fetching a project with the given project id fails",
+      () {
+        whenFetchProject().thenAnswer((_) => Future.error(firestoreException));
+
+        final result = adapter.updateBuilds(testProjectId, []);
+
+        expect(result, throwsDestinationError);
+      },
+    );
+
+    test(
+      ".updateBuilds() throws an ArgumentError if a project with the given id does not exist",
+      () {
+        whenCheckProjectExists().thenReturn(false);
+
+        final result = adapter.updateBuilds(testProjectId, []);
+
+        expect(result, throwsArgumentError);
+      },
+    );
+
+    test(
+      ".updateBuilds() does not update builds if the project with the given project id does not exist",
+      () {
+        whenCheckProjectExists().thenReturn(false);
+
+        final result = adapter.updateBuilds(testProjectId, builds);
+
+        expect(result, throwsArgumentError);
+        verifyNever(_firestoreMock.collection('build'));
+      },
+    );
+
+    test(
+      ".updateBuilds() references builds using their ids",
+      () async {
+        whenCheckProjectExists().thenReturn(true);
+        whenReferenceBuild().thenReturn(_documentReferenceMock);
+        when(_documentReferenceMock.update(any)).thenAnswer(
+          (_) => Future.sync(() {}),
+        );
+
+        await adapter.updateBuilds(testProjectId, builds);
+
+        verifyInOrder(
+          builds.map((build) {
+            _collectionReferenceMock.document(build.id);
+          }).toList(),
+        );
```

Comment:
Let's rewrite to make it a bit more readable: 
```suggestion
        verifyInOrder([
          for (final build in builds)
            _collectionReferenceMock.document(build.id),
        ]);
```
So the reader won't be confused about verifying the `map` is called.